axum/CHANGELOG.md

```diff
@@ -237,18 +237,48 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
       }
   }
   ```
-- **breaking:** The following types or traits have a new `S` type param
-  (`()` by default) which represents the state ([#1155]):
-  - `FromRequest`
-  - `RequestParts`
-  - `Router`
-  - `MethodRouter`
-  - `Handler`
+- **breaking:** It is now only possible for one extractor per handler to consume
+  the request body. In 0.5 doing so would result in runtime errors but in 0.6 it
+  is a compile error ([#1272])
+
+  axum enforces this by only allowing the _last_ extractor to consume the
+  request.
+
+  For example:
+
+  ```rust
+  use axum::{Json, http::HeaderMap};
+
+  // this wont compile on 0.6 because both `Json` and `String` needs to consume
+  // the request body. You must use either `Json` or `String` but not both.
```

Comment:
```suggestion
  // This wont compile on 0.6 because both `Json` and `String` need to consume
  // the request body. You can use either `Json` or `String`, but not both.
```